judge/management/commands/runmoss.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import print_function
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from moss import *
+
+from judge.models import *
+
+
+class Command(BaseCommand):
+    help = 'Checks for duplicate code using MOSS'
+
+    LANG_MAPPING = {
+        ('C++', MOSS_LANG_CC),
+        ('C', MOSS_LANG_C),
+        ('Java', MOSS_LANG_JAVA),
+        ('Python', MOSS_LANG_PYTHON),
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument('contest', help='the id of the contest')
+
+    def handle(self, *args, **options):
+        moss_api_key = getattr(settings, 'MOSS_API_KEY')
+        if moss_api_key is None:
+            print('No MOSS API Key supplied')
+            return
+        contest = options['name']
+
+        for problem in Contest.objects.get(key=contest).problems.order_by('code'):
+            print('========== %s / %s ==========' % (problem.code, problem.name))
+            for dmoj_lang, moss_lang in lang_mapping:
+                print("%s: " % dmoj_lang,end=' ')
+                subs = Submission.objects.filter(contest__participation__virtual=0, contest__participation__contest__key=contest,
+                                                 result='AC', problem__id=problem.id, language__common_name=dmoj_lang).values_list('user',
+                                                 'source')
+                if not subs:
+                    print('<no submissions>')
+                    continue
+
+                moss_call = MOSS(moss_api_key, language=moss_lang, matching_file_limit=100, comment='%s - %s' % (contest, problem.code))
+
+                users = set()
+
+                for sub in subs:
+                    if sub[0] in users:
+                        continue
+                    users.add(sub[0])
+                    moss_call.add_file_from_memory(sub[0].user.username, sub[1].encode('ascii', 'replace'))
```

Comment:
Wait why are you encoding as ASCII????? Someone just has to put in é in a comment to crash...